runtimevar/blobvar/blobvar.go

```diff
@@ -34,6 +51,90 @@ import (
 	"gocloud.dev/runtimevar/driver"
 )
 
+func init() {
+	runtimevar.DefaultURLMux().RegisterVariable(Scheme, &lazyBucketOpener{})
+}
+
+// Scheme is the URL scheme blobvar registers its URLOpener under on runtimevar.DefaultMux.
+const Scheme = "blob"
+
+type lazyBucketOpener struct {
+	mu      sync.Mutex
+	buckets map[string]*blob.Bucket
+}
+
+func (o *lazyBucketOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	bucketURL := u.Query().Get("bucket")
+	if bucketURL == "" {
+		return nil, fmt.Errorf("open variable %q: URL parameter \"bucket\" is required", u)
+	}
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	if o.buckets == nil {
+		o.buckets = map[string]*blob.Bucket{}
+	}
+	bucket := o.buckets[bucketURL]
+	if bucket == nil {
+		var err error
+		bucket, err = blob.OpenBucket(ctx, bucketURL)
+		if err != nil {
+			return nil, fmt.Errorf("open variable %q: OpenBucket failed: %v", u, err)
+		}
+		o.buckets[bucketURL] = bucket
```

Comment:
It would be nice if these buckets would get reclaimed as their variables get closed, but I can't see this causing huge problems. Mind opening an issue to track?

Reply:
Done: #1485.